libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+                    recognizerResult,
+                    luisModel = new
+                    {
+                        ModelID = _application.ApplicationId,
+                    },
+                    luisOptions = options,
+                    luisResult = luisResponse,
+                });
+
+            await turnContext.TraceActivityAsync("LuisV3Recognizer", traceInfo, LuisTraceType, LuisTraceLabel, cancellationToken).ConfigureAwait(false);
+            return recognizerResult;
+        }
+
+        private LuisV3PredictionOptions MergeDefaultOptionsWithProvidedOptions(LuisV3PredictionOptions defaultOptions, LuisV3PredictionOptions overridenOptions)
+            => new LuisV3PredictionOptions()
+            {
+                DatetimeReference = overridenOptions.DatetimeReference ?? defaultOptions.DatetimeReference,
+                DynamicLists = overridenOptions.DynamicLists ?? defaultOptions.DynamicLists,
+                ExternalEntities = overridenOptions.ExternalEntities ?? defaultOptions.ExternalEntities,
+                IncludeAllIntents = overridenOptions.IncludeAllIntents ?? defaultOptions.IncludeAllIntents,
+                IncludeInstanceData = overridenOptions.IncludeInstanceData ?? defaultOptions.IncludeInstanceData,
+                Log = overridenOptions.Log ?? defaultOptions.Log,
+                PreferExternalEntities = overridenOptions.PreferExternalEntities ?? defaultOptions.PreferExternalEntities,
+                Slot = overridenOptions.Slot ?? defaultOptions.Slot,
+                Version = overridenOptions.Version ?? defaultOptions.Version,
+            };
+
+        private DelegatingHandler CreateHttpHandlerPipeline(HttpClientHandler httpClientHandler, params DelegatingHandler[] handlers)
+        {
+            // Now, the RetryAfterDelegatingHandler should be the absolute outermost handler
+            // because it's extremely lightweight and non-interfering
+            DelegatingHandler currentHandler =
+                new RetryDelegatingHandler(new RetryAfterDelegatingHandler { InnerHandler = httpClientHandler });
+
+            if (handlers != null)
+            {
+                for (var i = handlers.Length - 1; i >= 0; --i)
+                {
+                    var handler = handlers[i];
+
+                    // Non-delegating handlers are ignored since we always
+                    // have RetryDelegatingHandler as the outer-most handler
+                    while (handler.InnerHandler is DelegatingHandler)
+                    {
+                        handler = handler.InnerHandler as DelegatingHandler;
+                    }
+
+                    handler.InnerHandler = currentHandler;
+                    currentHandler = handlers[i];
+                }
+            }
+
+            return currentHandler;
+        }
+
+        private HttpClientHandler CreateRootHandler() =>
+
+            // Create our root handler
+#if FullNetFx
+            return new WebRequestHandler();
```

Comment:
Is this still needed? I don't understand the difference between these two...

Reply:
I've removed it in both versions--no idea why it is there.